### PR TITLE
Use new extension syntax in scala.tasty.Reflection

### PR DESCRIPTION
**Pull request by nicolasstucki**
Extension are placed within the module of the type. For example, the extension `Tree.pos`
will be available explicitly and in the documentation as `Tree.extension_pos`.
We still need to add an `given XYZOps as XYZ.type = XYZ` to make the extension methods available.

### CODE REVIEW

File: library/src/scala/tasty/Reflection.scala [hunk 1]
```diff
@@ -423,37 +423,41 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
   def rootContext: Context = internal.rootContext // TODO: Should this be moved to QuoteContext?
   given Context = rootContext // TODO: Should be an implicit converion from QuoteContext to Context
 
-  extension ContextOps on (self: Context) {
-    /** Returns the owner of the context */
-    def owner: Symbol = internal.Context_owner(self)
+  given ContextOps as Context.type = Context
```
Comment (liufengyun) [inline library/src/scala/tasty/Reflection.scala:426]: This seems to suggest something is wrong with the language design.

File: library/src/scala/tasty/Reflection.scala [hunk 2]
```diff
@@ -423,37 +423,41 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
...
 
-    /** Returns the source file being compiled. The path is relative to the current working directory. */
-    def source: java.nio.file.Path = internal.Context_source(self)
+  object Context:
+    extension (self: Context):
+      /** Returns the owner of the context */
+      def owner: Symbol = internal.Context_owner(self)
```
Comment (liufengyun) [inline library/src/scala/tasty/Reflection.scala:431]: The previous extension methods syntax seems much better for this use case.

File: library/src/scala/tasty/Reflection.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala:426]: It is more about the automatically imported scopes for abstract types. With opaque types we look into the module but for these ones we are missing it. I plan on minimizing this limitation and ideally remove those givens later

**Comment by nicolasstucki** [comment]
@anatoliykmetyuk could you should review the changes in the community build. I removed the `TraceHelper` and all the useless imports it added. That pattern should never be used.

**Pull request merged**